### DISCUSSION TITLE
The prompt given in node 14.17.3 is Cannot convert undefined or null to object

### DISCUSSION THREAD

**Discussion by dayney** [Zhengqbbb/cz-git#198] [Q&A]
These errors were displayed when I reinstalled：
```
npm WARN deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.

> yikuaiqian@2.1.1 prepare /Users/krui/git/work/purchase-weapp
> husky install

husky - Git hooks installed
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN notsup Unsupported engine for cosmiconfig-typescript-loader@4.4.0: wanted: {"node":">=v14.21.3"} (current: {"node":"14.17.3","npm":"6.14.13"})
npm WARN notsup Not compatible with your version of node/npm: cosmiconfig-typescript-loader@4.4.0
```

after, I run the following command to give the prompt is
```
git add package*
git cz 
```

```
Cannot convert undefined or null to object
```
在node20中可以正常使用，这个问题需要怎么样解决

**Comment by Zhengqbbb** [comment]
what version?
```sh
npx envinfo --system --binaries --npmPackages cz-git --npmPackages commitizen --npmPackages czg --npmGlobalPackages cz-git --npmGlobalPackages commitizen --npmGlobalPackages czg
```

---

> npm notice created a lockfile as package-lock.json. You should commit this file.
> npm WARN notsup Unsupported engine for cosmiconfig-typescript-loader@4.4.0: wanted: {"node":">=v14.21.3"} (current: >>{"node":"14.17.3","npm":"6.14.13"})
> npm WARN notsup Not compatible with your version of node/npm: cosmiconfig-typescript-loader@4.4.0

cz-git and czg do not have this dependency `cosmiconfig-typescript-loader`, please check

**Comment by Zhengqbbb** [comment]
> Cannot convert undefined or null to object

I have reproduced the issue, which is consistent with my suspicion. In `v1.9.4` - https://github.com/Zhengqbbb/cz-git/blob/main/CHANGELOG.md#194-2024-07-19, although the error for node v22 has been fixed, it introduced an incompatible error. I apologize for the inconvenience; please downgrade to `v1.9.3` first.


```sh
npm i -D cz-git@1.9.3
```

**Comment by Zhengqbbb** [comment]
Follow ISSUE #199
